### PR TITLE
1475 contributor dates

### PR DESCRIPTION
**Pull request by TobiasNx**
I added the dates of `contribution.agent` and RWSK subjects:

`dateOfBirth`
`dateOfDeath`
`dateOfBirthAndDeath`

### CODE REVIEW

File: src/test/resources/alma-fix/990174080470206441.json [hunk 1]
```diff
@@ -0,0 +1,297 @@
+{
```
Comment (dr0i) [inline src/test/resources/alma-fix/990174080470206441.json:1]: Not sure whether introducing of this (and other) documents is superflous as the old test files already cover the case.

File: src/test/resources/alma-fix/990030574430206441.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  "type" : [ "BibliographicResource", "Book" ],
+  "responsibilityStatement" : [ "testi per uso accademico. A cura di F[rancesco] Sbordone" ],
+  "contribution" : [ {
+    "agent" : {
+      "gndIdentifier" : "(DE-588)118506080",
+      "id" : "https://d-nb.info/gnd/118506080",
+      "label" : "Bacchylides",
+      "type" : [ "Person" ],
+      "dateOfBirth" : "-0516",
+      "dateOfDeath" : "-0451"
```
Comment (dr0i) [inline src/test/resources/alma-fix/990030574430206441.json:85]: To add this document is ok because it covers an edge case (negative year).

File: src/test/resources/alma-fix/990030574430206441.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      "label" : "Autor/in"
+    },
+    "type" : [ "Contribution" ]
+  }, {
+    "agent" : {
+      "gndIdentifier" : "(DE-588)118594427",
+      "id" : "https://d-nb.info/gnd/118594427",
+      "label" : "Pindarus",
+      "type" : [ "Person" ],
+      "dateOfBirthAndDeath" : "ca. 522 oder 518 v. Chr.-446 v. Chr."
```
Comment (dr0i) [inline src/test/resources/alma-fix/990030574430206441.json:98]: and it also covers the newly introduced "dateOfBirthAndDeath".

File: src/test/resources/alma-fix/990210950050206441.json [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    "componentList" : [ {
+      "type" : [ "Person" ],
+      "label" : "Ferd",
+      "source" : {
+        "label" : "Gemeinsame Normdatei (GND)",
+        "id" : "https://d-nb.info/gnd/7749153-1"
+      },
+      "id" : "http://d-nb.info/gnd/1093886471",
+      "gndIdentifier" : "1093886471",
+      "dateOfBirthAndDeath" : "ca. 20.-21. Jh."
```
Comment (dr0i) [inline src/test/resources/alma-fix/990210950050206441.json:125]: already covered?

File: src/test/resources/alma-fix/990150856900206441.json [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  "type" : [ "BibliographicResource", "EditedVolume", "Book" ],
+  "responsibilityStatement" : [ "von Hans Lietzmann" ],
+  "contribution" : [ {
+    "agent" : {
+      "gndIdentifier" : "(DE-588)118572865",
+      "id" : "https://d-nb.info/gnd/118572865",
+      "label" : "Lietzmann, Hans",
+      "type" : [ "Person" ],
+      "dateOfBirth" : "1875",
+      "dateOfDeath" : "1942"
```
Comment (dr0i) [inline src/test/resources/alma-fix/990150856900206441.json:105]: already covered?

**Review by dr0i — CHANGES_REQUESTED** [review]
Added some comments. It's about "are we introducing test documents which are superflous?"  I didn't have had a lokk into the XML so it may be that they are not superflous but cover different source fields. Do they?

**Review by acka47 — APPROVED** [review]
+1

File: src/test/resources/alma-fix/990150856900206441.json (hunk 5 above)
Comment (TobiasNx) [reply src/test/resources/alma-fix/990150856900206441.json:105]: The edge case is the second contribution object which adds zeros at the beginning of years that have less than 4 digits.

File: src/test/resources/alma-fix/990150856900206441.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    "type" : [ "Contribution" ]
+  }, {
+    "agent" : {
+      "gndIdentifier" : "(DE-588)118649817",
+      "id" : "https://d-nb.info/gnd/118649817",
+      "label" : "Apollinaris",
+      "type" : [ "Person" ],
+      "dateOfBirth" : "0310",
+      "dateOfDeath" : "0390"
```
Comment (TobiasNx) [inline src/test/resources/alma-fix/990150856900206441.json:119]: These are the edge case

File: src/test/resources/alma-fix/990030574430206441.json (hunk 3 above)
Comment (TobiasNx) [reply src/test/resources/alma-fix/990030574430206441.json:98]: Here it is if the filter works with ca. alone.

File: src/test/resources/alma-fix/990210950050206441.json (hunk 4 above)
Comment (TobiasNx) [reply src/test/resources/alma-fix/990210950050206441.json:125]: This filters to ca. and Jh. different case as the other.

File: src/test/resources/alma-fix/990174080470206441.json (hunk 1 above)
Comment (TobiasNx) [reply src/test/resources/alma-fix/990174080470206441.json:1]: This is the only test file that does not introduce any new info. But should we keep it so that I do not spent to much time rebuilding the branch?

**Comment by TobiasNx** [comment]
@dr0i only one test files introduces no new test case. I would suggest to keep it.
Since rebuilding the branch would be to time consuming and all test are borrowed from our old JSONLD Aleph tests.

**Comment by dr0i** [comment]
Which one is it? I would kick it out - anyway we need to solve the conflict.

**Comment by dr0i** [comment]
Ah, you noted it already: [990174080470206441.json](https://github.com/hbz/lobid-resources/pull/1503/files/71b3411e675ba1b3d159d027b7da12a872620f8e#diff-f82f3aa5bab0e8e18c3dccd0135023c71f19d6a00a0a55adcaf49e8569e1b485)
